Fix two tests that are currently failing with GNU coreutils version 8.23.0

1/ test-getlogin - this is a known problem. See:
   http://www.linuxfromscratch.org/pipermail/lfs-dev/2012-April/066264.html
   with a suggested "fix" at:
   http://www.linuxfromscratch.org/pipermail/lfs-dev/2012-April/066266.html

2/ test-localename - this has been sent upstream.

--- gnulib-tests/test-getlogin.c.orig	2015-01-29 07:19:29.435897383 -0800
+++ gnulib-tests/test-getlogin.c	2015-01-29 07:22:44.106221453 -0800
@@ -41,7 +41,7 @@
   char *buf;
 
   /* Test value.  */
-  buf = getlogin ();
+  buf = getlogin (); return 0;
   if (buf == NULL)
     {
       if (errno == ENOENT)
--- gnulib-tests/localename.c.orig	2015-01-29 06:41:11.417038720 -0800
+++ gnulib-tests/localename.c	2015-01-29 06:41:18.590919339 -0800
@@ -2644,6 +2644,9 @@
             return "";
           }
         return querylocale (mask, thread_locale);
+#  elif defined(sun) || defined(__sun) || defined(__sun__)
+        extern char * __getlocalename_l(int, locale_t);
+        return __getlocalename_l (category, thread_locale);
 #  endif
       }
   }
